AssignResourceChangeStatus.cs

```diff
@@ -112,11 +80,42 @@ public IssueDto CopyIssueComplete(IssueDtoEventArgs args)
 
         public IssueDto BeforeCreateFull(IssueDtoEventArgs args)
         {
-            return args.Issue;
+            return UpdateStatus( args );
         }
 
         public IssueDto BeforeUpdateFull(IssueDtoEventArgs args)
         {
+            return UpdateStatus(args);
+        }
+
+        private IssueDto UpdateStatus( IssueDtoEventArgs args )
+        {
+            //If there is no resource, then do not want to do anything
+            if ( args.Issue.Entity.GetResources().Count == 0 ) return args.Issue;
+
+            var project = args.Context.Projects.Get( args.Issue.Entity.ProjectId );
+
+            if ( project == null ) return args.Issue;
+
+            var metaManager = GeminiApp.GetManager<MetaManager>( args.User );
+
+            var issueType = metaManager.TypeGet( args.Issue.Entity.TypeId ) ; // args.Context.Meta.TypeGet( args.Issue.Entity.TypeId );
+            var issueStatus = metaManager.StatusGet( args.Issue.Entity.StatusId ); // args.Context.Meta.StatusGet( args.Issue.Entity.StatusId );
+
+            var workflow = metaManager.StatusGetWorkflow( issueType.Entity, issueStatus.Entity, args.Issue.Entity )
+                .OrderBy( s => s.Entity.Sequence )
```

Comment:
Why are we doing order by after getting workflow?

Reply:
was to ensure it was in the right order, but investigating and can see it should be in the right order